src/autofix/solutions/codeReplacer.ts

```diff
@@ -0,0 +1,321 @@
+		const patchedFixHints = patchMessageFixHints(fixHints, apiName ?? functionName);
+
+		if (!patchedFixHints || !("exportCodeToBeUsed" in patchedFixHints) ||
+			!patchedFixHints.exportCodeToBeUsed || !(typeof patchedFixHints.exportCodeToBeUsed === "object") ||
+			!position ||
+			!moduleDeclarations.size) {
+			continue;
+		}
+
+		// Find the imports from module declarations in the context/scope of the message
+		const importRequests = getContextImportRequests(sourceFile, moduleDeclarations, position);
+
+		const {exportCodeToBeUsed, moduleName} = patchedFixHints;
+		const moduleInfo = moduleName ? importRequests.get(moduleName) : null;
+		if (moduleInfo?.identifier) {
+			exportCodeToBeUsed.name =
+				exportCodeToBeUsed.name.replaceAll("$moduleIdentifier", moduleInfo.identifier);
+		}
+		let identifierIndex = 1;
+		while (exportCodeToBeUsed.name.includes("$identifier")) {
+			const identifier = resolveUniqueName(apiName ?? moduleName ?? "j", declaredIdentifiers);
+			declaredIdentifiers.add(identifier);
+			exportCodeToBeUsed.name =
+				exportCodeToBeUsed.name.replaceAll(`$identifier_${identifierIndex}`, identifier);
+			identifierIndex++;
+		}
+
+		let value = exportCodeToBeUsed.args?.reduce((acc, arg, index) => {
+			return acc?.replace(new RegExp(`\\$${index + 1}(?!\\d)`, "g"), arg.value);
+		}, exportCodeToBeUsed.name ?? "") ?? exportCodeToBeUsed.name;
+		value = value.replaceAll(/\$\d+/g, "undefined"); // Some placeholders might be "empty" in the original code
+
+		// Calculate the replacement position
+		// It cannot be derived from the fixHintsGenerator as it works on the compiled source file.
+		// Just take the length of the old and the original start position
+		let pos;
+		if (moduleInfo?.nodeInfos) {
+			// For jQuery function deprecations, the node might only cover
+			// the deprecated method, but not the whole expression.
+			// Find the expression and update the position.
+			const nodeInfoAtPosition = moduleInfo.nodeInfos.find((nodeInfo) => {
+				return nodeInfo.position.line === (position.line - 1) &&
+					nodeInfo.position.column === (position.column - 1);
+			});
+
+			if (nodeInfoAtPosition?.node && !ts.isCallExpression(nodeInfoAtPosition.node) &&
+				ts.isPropertyAccessExpression(nodeInfoAtPosition.node) &&
+				nodeInfoAtPosition.node.expression && ts.isCallExpression(nodeInfoAtPosition.node.expression)) {
+				pos = sourceFile.getLineAndCharacterOfPosition(nodeInfoAtPosition.node.expression.pos);
+			}
+		}
+		const line = pos ? pos.line : (position.line - 1);
+		const column = pos ? (pos.character + 1) : (position.column - 1);
+		const start = sourceFile.getPositionOfLineAndCharacter(line, column);
+		const end = start + exportCodeToBeUsed.solutionLength;
+
+		cleanupChangeSet(changeSet, start, end);
+
+		changeSet.push({
+			action: ChangeAction.REPLACE,
+			start,
+			end,
+			value,
+		});
+	}
+}
+
+const importRequestsCache = new Map<number, ImportRequests>();
```

Comment:
This is a singleton cache that is not invalidated across runs. It will most likely cause side-effects when running multiple autofix runs in the same process.

Reply:
I might refactor this code anyways, as we need similar functionality somewhere else.

Reply:
Done